pkg/datastore/mysql/iterator.go

```diff
@@ -35,10 +40,89 @@ func (it *Iterator) Next(dst interface{}) error {
 	if err != nil {
 		return err
 	}
+
+	// Update last iterated item as last read row.
+	it.last = &rowDataConverter{val: val}
+
 	return decodeJSONValue(val, dst)
 }
 
-// Cursor implementation for MySQL Iterator
+// Cursor builds a base 64 string (encode from string in map[string]interface{} format).
+// The cursor contains only values attached with the fields used
+// as ordering fields.
 func (it *Iterator) Cursor() (string, error) {
-	return "", datastore.ErrUnimplemented
+	if it.last == nil {
+		return "", datastore.ErrInvalidCursor
+	}
+
+	lastObjData := it.last.Data()
+
+	cursor := make(map[string]interface{}, len(it.orders))
+	for _, o := range it.orders {
+		val, ok := lastObjData[o.Field]
+		if !ok {
+			return "", datastore.ErrInvalidCursor
+		}
+		// TODO: Support build cursor from nested Ordering field.
+		cursor[o.Field] = val
+	}
+
+	b, _ := json.Marshal(cursor)
+	return base64.StdEncoding.EncodeToString(b), nil
+}
+
+type DataConverter interface {
+	Data() map[string]interface{}
+}
+
+type rowDataConverter struct {
+	val string
+}
+
+// Data make JSON object with key in CamelCase format.
+func (r *rowDataConverter) Data() map[string]interface{} {
+	jsonRaw := convertKeys(json.RawMessage(r.val), convertSnakeToCamel)
```

Comment:
While I'm in the middle of looking over the entire code, I'm just curious about why we need to convert fields in a snake case to one in a camel-case.

Reply:
It looks like `makePaginationCursorValues` retrieves cursor values by using Camel case (the value of o.Field will be supposed to Camel case right?)
```go
val, ok := obj[o.Field]
```
https://github.com/pipe-cd/pipe/pull/1873/files#diff-c47f267fba561fafc1be077abac3f330be6605efdd9290475d39cfe1a055fef0R250

Actually I don't even know what format the data will be saved in MySQL, so let me know if I was wrong.

Reply:
The `Data` field stored in MySQL datastore is using models object which autogenerated by gRPC code gen. It has `snake_case` JSON key which makes data stored in MySQL database is JSON with `snake_case` key 🐍 
As for now, our Filters and Orders are all written in `CamelCase` 🐫  - for example [web_api.go](https://github.com/pipe-cd/pipe/blob/master/pkg/app/api/grpcapi/web_api.go#L150) ( the cause is previously we only focus on Firestore, which do not use JSON but scan our object itself which makes data stored in Object format with CamelCase key as map[string]interface{} type).
What I tried to do there is to keep the Filters and Orders we wrote works, data values got from MySQL datastore have to build with `CamelCase` in JSON format so that Filters & Orders keys could be found in converted map data 👀

Reply:
> Actually I don't even know what format the data will be saved in MySQL

The data stored in MySQL Data field is using `json.Marshall` which use JSON keys autogenerated by gRPC, which means: they are JSON with `snake_case` key 😁 
https://github.com/pipe-cd/pipe/blob/e2d2dffe0ce989411b8f47ffad1e589c3b7557fa/pkg/datastore/mysql/model_wrapper.go#L100-L110

Reply:
@khanhtc1202 Thank you for telling me a lot!

>It has snake_case JSON key which makes data stored in MySQL database is JSON with snake_case key 

I haven't looked up it but, what I came up with is can we change it to camel case? I mean can we add a json tag kind of like `json="pipedId"` for instance:

```go
type Application struct {
	PipedId string `protobuf:"bytes,4,opt,name=PipedId,json=piped_id,proto3" json:"pipedId,omitempty"`
```

Reply:
I looked at some places and I knew it, a bunch of places already uses this json tag so I couldn't. Never mind.